### PR TITLE
Reactify revocation input step

### PR DESCRIPTION
**Pull request by frederike-ramin**
# Short Description
- See title

# Changes
- Add react component for revocation page
- Add it to storybook
- Use it in the step
- Fix the display of labels with examples (before, the class name was set but no css was defined for it)

# Feedback
- Do you see anything suspicious?
- Do you think I should just set the title and intro either a) directly in React or b) in the call of RevocationProps?

### CODE REVIEW

File: webapp/client/src/lib/translations.js [hunk 1]
```diff
@@ -95,6 +95,14 @@ const translations = {
...
     },
   },
+  unlockCodeRevocation: {
+    idnr: {
+      labelText: "Steuer-Identifikationsnummer",
+    },
+    dob: {
+      labelText: "Geburtsdatum",
+    },
+  },
```
Comment (JulianRoesner) [inline webapp/client/src/lib/translations.js:105]: I am just wondering if we really should split the translations in that way. Wouldn't it make more sense to set the label for the IDNr only once and reuse it for all the different flows. If it changes for the unlockCodeInput, it should also change here, shouldn't it?

File: webapp/client/src/pages/RevocationPage.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+export default function RevocationPage({ stepHeader, form, fields }) {
+  const { t } = useTranslation();
+
+  return (
+    <>
+      <StepHeaderButtons />
+      <FormHeader {...stepHeader} />
+      <StepForm {...form}>
+        <FormRowCentered>
+          <FormFieldIdNr
```
Comment (JulianRoesner) [inline webapp/client/src/pages/RevocationPage.js:21]: I know that this is not that important and it was part of the old version as well but why don't we use the detail here but use it on all the other pages? Maybe, we could talk 1 min. to Tine/Nadine and then add. Or we leave as is and file a "bug ticket" to prioritize it?

**Review by JulianRoesner — CHANGES_REQUESTED** [review]
Not to be too stubborn but why did you decide against adding functional tests before exchanging the old template with the react component?
Especially, because we do not test revocation in the acceptance tests we should test it with the functional tests (running against the mockErica?)

Other than that, it looks good

File: webapp/client/src/lib/translations.js (hunk 1 above)
Comment (frederike-ramin) [reply webapp/client/src/lib/translations.js:105]: Mh..maybe. In that case it should be directly set on the IdNr field then though to be consistent, I think? And then, when we do that, it's also only consistent to do that with all fields. And I'm not sure we want a PrenameField and a SurnameField etc.? But maybe thats. too dogmatic of me^^

File: webapp/client/src/pages/RevocationPage.js (hunk 2 above)
Comment (frederike-ramin) [reply webapp/client/src/pages/RevocationPage.js:21]: I think I actually know why. If you landed here, you already at least registered so you know what you're IdNr ist. But maybe that's wrong and I'm gonna quickly ask Nadine.

Comment (frederike-ramin) [reply webapp/client/src/pages/RevocationPage.js:21]: I asked her, she said, "Yes please thank you" - so another thanks to you^^

**Comment by frederike-ramin** [comment]
> Not to be too stubborn but why did you decide against adding functional tests before exchanging the old template with the react component?

Well.....because...I simply forgot. 🙈 Thanks for pointing that out!

**Comment by frederike-ramin** [comment]
I had a look at SonarCloud and everything there is okay to me. We have Babel and a necessary TODO comment and the duplicated lines are supposed to be duplicated. So I'm just gonna ignore that when merging^^

File: webapp/client/src/lib/translations.js (hunk 1 above)
Comment (JulianRoesner) [reply webapp/client/src/lib/translations.js]: Maybe, I did get your answer wrong but I think that we have a misunderstanding. I think there are two ways to think about it:
1.  Set it once in a specific component
2. Set it only once in the translations and do not connect it with a specific flow

I meant the second option. That would basically mean to change translations like this (or without the components) and then use that identifier anywhere we use a idnr field:
```
 components: {
    idnr: {
      labelText: "Steuer-Identifikationsnummer",
    },
    dob: {
      labelText: "Geburtsdatum",
    },
  },
```

File: webapp/client/src/pages/RevocationPage.js [hunk 3]
```diff
@@ -27,6 +27,10 @@ export default function RevocationPage({ stepHeader, form, fields }) {
             label={{
               text: t("unlockCodeRevocation.idnr.labelText"),
             }}
+            details={{
+              title: t("unlockCodeRevocation.idnr.help.title"),
+              text: t("unlockCodeRevocation.idnr.help.text"),
+            }}
```
Comment (JulianRoesner) [inline webapp/client/src/pages/RevocationPage.js:33]: I am really not sure if we need to set this here or if these should be part of the IDNr and could be toggled on and off, but that's maybe a discussion for another time 🤷

File: webapp/client/cypress/integration/revocation.spec.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      month: "2",
+      year: "1951",
+    };
+
+    beforeEach(() => {
+      cy.register(idnr, dob);
+      cy.visit("/unlock_code_revocation/step/data_input");
+    });
+
+    it("should redirect to success page", function () {
```
Comment (JulianRoesner) [inline webapp/client/cypress/integration/revocation.spec.js:66]: I did not understand it correctly. Maybe we could rename it to:
```
    it("should redirect to success page for correct input", function () {
```

File: webapp/tests/elster_client/mock_erica.py [hunk 5]
```diff
@@ -255,6 +255,7 @@ def revoke_unlock_code(input_body, show_response: bool):
         # Successful case
         if idnr_exists:
             elster_request_id_for_revocation = gen_random_key()
+            MockErica.available_idnrs = [idnr for idnr in MockErica.available_idnrs if idnr[0] != input_data.get('idnr')]
```
Comment (JulianRoesner) [inline webapp/tests/elster_client/mock_erica.py:258]: That is all that you had to change? Awesome!

File: webapp/client/cypress/integration/revocation.spec.js (hunk 4 above)
Comment (JulianRoesner) [reply webapp/client/cypress/integration/revocation.spec.js:66]: Somehow I did not connect the test correctly with the context and just thought about it as being automatically entered and not send.

**Review by JulianRoesner — APPROVED** [review]
LGTM! Nice work with the tests. Just some minor comments up to you

File: webapp/client/src/pages/RevocationPage.js (hunk 3 above)
Comment (frederike-ramin) [reply webapp/client/src/pages/RevocationPage.js:33]: Yes. And I think that's the same discussion as with the [comment](https://github.com/digitalservice4germany/steuerlotse/pull/367#discussion_r755020047) above. We have some texts that just always belong to one component. Question is if that coincidence and we leave it like this or if it's desired and we pull the messages into the component. Let's talk about it?

File: webapp/client/src/lib/translations.js (hunk 1 above)
Comment (frederike-ramin) [reply webapp/client/src/lib/translations.js]: I called moved it into `fields` now as that already existed. I suppose that's fine for you?

**Pull request merged**